pkg/providers/equinix.go

```diff
@@ -0,0 +1,121 @@
+package providers
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/google/uuid"
+	"github.com/packethost/packngo" //requires env var PACKET_AUTH_TOKEN
+)
+
+const (
+	hostPrefix string = "cir"
+)
+
+type equinixProviderConfig struct {
+	ProjectID string   `json:"projectid"` //project id in equinix
+	Facility  []string `json:"facility"`  //datacenters to use
+	Plan      string   `json:"plan"`      //server size
+	OS        string   `json:"os"`        //OS to install
+}
+
+type equinixProvider struct {
+	config equinixProviderConfig
+	client *packngo.Client
+}
+
+func EquinixProviderFactory(providerInfo string, secretData map[string][]byte) (Provider, error) {
+	config := equinixProviderConfig{
+		ProjectID: "replace-with-project-id",
+		Facility:  []string{"da11", "dc10", "dc13", "ny5", "sv15"},
+		Plan:      "c3.small.x86",
+		OS:        "rocky_8",
+	}
+
+	if configJson, ok := secretData["config"]; ok {
+		if err := json.Unmarshal(configJson, &config); err != nil {
+			return nil, err
+		}
+	}
+
+	client, err := packngo.NewClient()
```

Comment:
This gets created @ every reconcile loop, wouldn't be better to have a singleton?